### ISSUE TITLE
support for windows ?

### ISSUE THREAD

**Issue by juju4** [neillturner/kitchen-ansible#131]
Hello,

seems support is missing.
windows box build with vagrant following
http://kitchen.ci/blog/test-kitchen-windows-test-flight-with-vagrant/

```
I, [2016-01-30T20:17:59.409701 #14169]  INFO -- default-windows-2012r2: -----> Creating <default-windows-2012r2>...
I, [2016-01-30T20:18:01.480750 #14169]  INFO -- default-windows-2012r2: Bringing machine 'default' up with 'virtualbox' provider...
I, [2016-01-30T20:18:01.608960 #14169]  INFO -- default-windows-2012r2: ==> default: Importing base box 'windows-2012r2'...
[...]
I, [2016-01-30T20:20:32.478191 #14169]  INFO -- default-windows-2012r2: Finished Preparing files for transfer
I, [2016-01-30T20:20:32.478330 #14169]  INFO -- default-windows-2012r2: Installing ansible, will try to determine platform os
W, [2016-01-30T20:20:32.568097 #14169]  WARN -- default-windows-2012r2: The command line is too long.
E, [2016-01-30T20:20:32.692438 #14169] ERROR -- default-windows-2012r2: Converge failed on instance <default-windows-2012r2>.
E, [2016-01-30T20:20:32.692541 #14169] ERROR -- default-windows-2012r2: ------Exception-------
E, [2016-01-30T20:20:32.692556 #14169] ERROR -- default-windows-2012r2: Class: Kitchen::ActionFailed
E, [2016-01-30T20:20:32.692577 #14169] ERROR -- default-windows-2012r2: Message: WinRM exited (1) for command: [            if [ ! $(which ansible) ]; then
              if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ]; then
                if ! [ grep -q 'Amazon Linux' /etc/system-release ]; then
                            if [ ! $(which ansible) ]; then
            yum install epel-release -y
                          rpm -ivh http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
[...]
```

It seems windows code is not there and it try the unix code.
Tested on Lubuntu 14.04 with ruby 1.9.3p484 
$ gem list 

*** LOCAL GEMS ***

builder (3.2.2)
faraday (0.9.2)
ffi (1.9.10)
gssapi (1.2.0)
gyoku (1.3.1)
highline (1.7.8)
httpclient (2.7.1)
kitchen-ansible (0.0.35)
kitchen-docker (2.3.0)
kitchen-vagrant (0.19.0)
librarian (0.1.2)
librarian-ansible (1.0.6)
little-plugger (1.1.4)
logging (2.0.0)
mixlib-shellout (2.2.5)
multi_json (1.11.2)
multipart-post (2.0.0)
net-scp (1.2.1)
net-ssh (2.9.2)
nori (2.6.0)
rubyntlm (0.5.3)
rubyzip (1.1.7)
safe_yaml (1.0.4)
test-kitchen (1.4.2)
thor (0.19.1)
winrm (1.6.0)
winrm-transport (1.0.3)

Thanks a lot for all the great work!

**Comment by neillturner** [comment]
The best way to do windows is to use a linux box to run ansible and serverspec via ssh from your workstation but then remotely config the windows  servers on a third box 
as per http://ec2dream.blogspot.co.uk/2015/12/automated-devops-with-test-kitchen.html

**Comment by juju4** [comment]
Hello Neil,

My host is a linux box and that's where kitchen is executed as a role test suite but from other role I tested, default behavior seems to converge the box with a locally installed ansible (which can be an issue on older distribution). So you mean that's my current problem, you mean?
In that case, how do I say to kitchen to converge and test with local tools to the remote guest ? (in my case local vagrant virtualbox/VM).
It would probably nice to trap shell code for windows platform anyway.

Here is my .kitchen.yml and default.yml
```
---
driver:
  name: vagrant

provisioner:
  name: ansible_playbook
  roles_path: ../../roles
  hosts: test-kitchen
#  ansible_verbose: true
  ansible_verbose: false
  ansible_verbosity: 3
  ansible_extra_flags: <%= ENV['ANSIBLE_EXTRA_FLAGS'] %>
  ansible_yum_repo: http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
  enable_yum_epel: true

platforms:
  - name: ubuntu-14.04
  - name: ubuntu-12.04
## FIXME! bash: ansible-playbook: command not found
#  - name: debian-6.0.10
#  - name: debian-7.0.
  - name: centos-7.1
#  - name: centos-6.6
## FIXME! The box failed to unpackage properly. Please verify that the box file you're trying to add is not corrupted and try again. The output from attempting to unpacka
ge (if any):
#  - name: centos-6.7
#  - name: centos-5.11
## http://kitchen.ci/blog/test-kitchen-windows-test-flight-with-vagrant/
## FIXME! https://github.com/neillturner/kitchen-ansible/issues/131
#  - name: windows-2012r2
#    transport:
#      name: winrm

suites:
  - name: default
    run_list:
    attributes:
```

```
- hosts: test-kitchen
  vars:
    - [...]
  roles:
    - testedrole
```

Thanks

**Comment by neillturner** [comment]
if you look at:    https://github.com/neillturner/ansible_repo by setting 
    ansible_connection: ssh 
rather than
    ansible_connection
ansible then it uses ssh to go the servers you specify in the hosts. 
ofcourse windows would need an sshd server on it to work.
 
we could also add windows support as per 
  http://docs.ansible.com/ansible/intro_windows.html 
for powershell remoting. 
and specify  
   ansible_connection: winrm
what we could do it take the omnibus ansible script from 
   ansible_omnibus_url
   https://raw.githubusercontent.com/neillturner/omnibus-ansible/master/ansible_install.sh
and add the extra install stuff for windows support 
 ie
   pip install "pywinrm>=0.1.1"
  Installing python-kerberos dependencies
  Installing python-kerberos
  Configuring Kerberos
as per http://docs.ansible.com/ansible/intro_windows.html
once this work i can move the install into the main code 
and add a parameter say
  windows_support: true

**Comment by juju4** [comment]
I would be more than happy to test. Powershell seems the way to go until Microsoft release its Windows with sshd.

is there a way to have the ansible_connection set by platform ? set automatically winrm if windows and ssh if unix. (Have also this issue with ansible_yum_repo which would depend on distribution release)

just made a quick test with ansible_connection: ssh in provisioner section of .kitchen.yml and I still got
```
       Transferring files to <default-debian-6010>
       Going to invoke ansible-playbook with: ANSIBLE_ROLES_PATH=/tmp/kitchen/roles sudo -Es  ansible-playbook --inventory-file=/tmp/kitchen/hosts -c ssh -M /tmp/kitchen/modules         /tmp/kitchen/default.yml
       /bin/bash: ansible-playbook: command not found
>>>>>> Converge failed on instance <default-debian-6010>.
```

**Comment by neillturner** [comment]
OK. I'll write the code for windows support this week and then you can install and test. in regard to settng the ansible_connection lets get it working first and then look at improvements like that.

**Comment by neillturner** [comment]
I have a version for testing - see the windows-support branch:

         https://github.com/neillturner/kitchen-ansible/tree/windows-support

download the kitchen-ansible-0.0.38.gem from the branch and install. 
The have tested the install but not running ansible with winrm. 
you need to specify: 

         ansible_connection:  winrm. 
         require_windows_support: true  
         kerberos_conf_file:   <your krb5.conf file path> 

later i will let the ansible_connection to be set to something like none which will not put on the 
ansible_connection parameter on the ansible cmd and then you can specify in the hosts files so you can have winrm and sssh hosts in the one ansible run.  

 when i get time i will test it with a windows server as well as the linux server running ansible.

**Comment by juju4** [comment]
Thanks a lot Neil. Here a quick feedback

is the kerberos_conf_file necessary? from ansible docs, it seems only when AD is involved which would not be the case for unit testing? (http://docs.ansible.com/ansible/intro_windows.html#configuring-kerberos)
what to put when there is no domain, just Workgroup?

with following .kitchen.yml
```
---
driver:
  name: vagrant

provisioner:
  name: ansible_playbook
  roles_path: ../../roles
  hosts: test-kitchen
  ansible_verbose: false
  ansible_verbosity: 3
  ansible_extra_flags: <%= ENV['ANSIBLE_EXTRA_FLAGS'] %>
  ansible_yum_repo: http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
  enable_yum_epel: true
  ansible_connection:  winrm
  require_windows_support: true  

platforms:
  - name: windows-2012r2
    transport:
      name: winrm

suites:
  - name: default
    run_list:
    attributes:
```


quick test. VM is not started
```
$ sudo -H -u deploy gem install --local /opt/tmp/vagrant/kitchen-ansible-0.0.38.gem --user-install
$ sudo -H -u deploy kitchen verify
-----> Starting Kitchen (v1.4.2)
-----> Converging <default-windows-2012r2>...
       Preparing files for transfer
       Preparing playbook
       Preparing inventory file
       Preparing modules
       nothing to do for modules
       Preparing roles
       Preparing ansible.cfg file
       Empty ansible.cfg generated
       Preparing group_vars
       nothing to do for group_vars
       Preparing additional_copy_path
       Preparing host_vars
       nothing to do for host_vars
       Preparing hosts file
       Preparing spec
       nothing to do for spec
       Preparing library plugins
       nothing to do for library plugins
       Preparing callback plugins
       nothing to do for callback plugins
       Preparing filter_plugins
       nothing to do for filter_plugins
       Preparing lookup_plugins
       nothing to do for lookup_plugins
       Finished Preparing files for transfer
       Installing ansible, will try to determine platform os
       [WinRM] connection failed, retrying in 1 seconds (#<Errno::ECONNREFUSED: Connection refused - connect(2) (127.0.0.1:5985)>)
       [WinRM] connection failed, retrying in 1 seconds (#<Errno::ECONNREFUSED: Connection refused - connect(2) (127.0.0.1:5985)>)
       [WinRM] connection failed, retrying in 1 seconds (#<Errno::ECONNREFUSED: Connection refused - connect(2) (127.0.0.1:5985)>)
       [WinRM] connection failed, retrying in 1 seconds (#<Errno::ECONNREFUSED: Connection refused - connect(2) (127.0.0.1:5985)>)
$$$$$$ [WinRM] connection failed, terminating (#<Errno::ECONNREFUSED: Connection refused - connect(2) (127.0.0.1:5985)>)
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #converge action: [Connection refused - connect(2) (127.0.0.1:5985)]
>>>>>> ----------------------
```
it does not seem to launch any VM or at least not visible in virtualbox like unix ones.

Thanks

**Comment by juju4** [comment]
My boad, seems a mess up vagrant status after a reboot... a kitchen destroy+verify solved it.

also need to run with user with gui. I was doing my testing with a separate user than my current login and didn't work.
When reverting to the one who has X11, worked... might miss some headless option. seems unlike unix, you need to say ```gui: false``` for windows platforms driver. Yes, that's it.

but it still tries to go unix shell...
```
$ kitchen verify
-----> Starting Kitchen (v1.4.2)
-----> Creating <default-windows-2012r2>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: Importing base box 'windows-2012r2'...
==> default: Matching MAC address for NAT networking...
       ==> default: Setting the name of the VM: kitchen-loki-default-windows-2012r2_default_1454863473809_59095
       ==> default: Clearing any previously set network interfaces...
       ==> default: Preparing network interfaces based on configuration...
           default: Adapter 1: nat
       ==> default: Forwarding ports...    default: 3389 => 3389 (adapter 1)
           default: 5985 => 5985 (adapter 1)
       ==> default: Running 'pre-boot' VM customizations...
       ==> default: Booting VM...==> default: Waiting for machine to boot. This may take a few minutes...    default: WinRM address: 127.0.0.1:5985
           default: WinRM username: vagrant
           default: WinRM transport: plaintext
       ==> default: Machine booted and ready!
       ==> default: Checking for guest additions in VM...
       ==> default: Machine not provisioned because `--no-provision` is specified.[WinRM] Established
       Vagrant instance <default-windows-2012r2> created.
       Finished creating <default-windows-2012r2> (2m55.19s).
-----> Converging <default-windows-2012r2>...
       Preparing files for transfer
       Preparing playbook
       Preparing inventory file
       Preparing modules
       nothing to do for modules
       Preparing roles
       Preparing ansible.cfg file
       Empty ansible.cfg generated
       Preparing group_vars
       nothing to do for group_vars
       Preparing additional_copy_path
       Preparing host_vars
       nothing to do for host_vars
       Preparing hosts file
       Preparing spec
       nothing to do for spec
       Preparing library plugins
       nothing to do for library plugins
       Preparing callback plugins
       nothing to do for callback plugins
       Preparing filter_plugins
       nothing to do for filter_plugins
       Preparing lookup_plugins
       nothing to do for lookup_plugins
       Finished Preparing files for transfer
       Installing ansible, will try to determine platform os
$$$$$$ The command line is too long.
>>>>>> Converge failed on instance <default-windows-2012r2>.
>>>>>> Please see .kitchen/logs/default-windows-2012r2.log for more details
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: WinRM exited (1) for command: [            if [ ! $(which ansible) ]; then
              if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ]; then
                if ! [ grep -q 'Amazon Linux' /etc/system-release ]; then
                            if [ ! $(which ansible) ]; then
            yum install epel-release -y
                          rpm -ivh http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm

            yum makecache
            yum -y install ansible libselinux-python git
            fi

                else
                            if [ ! $(which ansible) ]; then
              yum install epel-release -y
[...]
```

or is it because there is 2 version installed ?
```
$ sudo -H -u deploy gem list|grep kitchen
kitchen-ansible (0.0.38, 0.0.35)
```

user install has not precedence?

**Comment by neillturner** [comment]
You need a linux server to run the ansible with windows support so a .kitchen.yml file like this

```
---
driver:
  name: vagrant

provisioner:
  name: ansible_playbook
  roles_path: ../../roles
  hosts: test-kitchen
  ansible_verbose: false
  ansible_verbosity: 3
  ansible_extra_flags: <%= ENV['ANSIBLE_EXTRA_FLAGS'] %>
  ansible_yum_repo: http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
  enable_yum_epel: true
  ansible_connection:  winrm
  require_windows_support: true  

platforms:
  - name: centos-7.0
    driver_plugin: vagrant
    driver_config:
      box: centos-7.0
  - name: windows-2012r2

suites:
  - name: default
    run_list:
    attributes:
```

Then 

1. Create Servers 

    kitchen create default-centos-70 -l debug
    kitchen create default-windows-2012r2 -l debug

to create the linux anible control server 
and the windows server that is to be configured 

2. Update the hosts file with the IP address of the windows server.

3. install and run ansible.
   kitchen converge default-centos-70 -l debug

I would not worry about the kerberos_conf_files at the moment. 

It is like 
  http://ec2dream.blogspot.co.uk/2015/12/automated-devops-with-test-kitchen.html
but instead of the tomcat server we have a windows server. 
 
 When we have it working i can add a blog post about windows support.

**Comment by juju4** [comment]
I'm trying to reproduce your dual setup but doesn't seem good in full automated mode

with your existing example
```
$ git clone https://github.com/neillturner/ansible_repo.git
$ cd ansible_repo
$ kitchen verify
-----> Starting Kitchen (v1.4.2)
-----> Creating <ansible-centos-70>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: VirtualBox VM is already running.
       [SSH] Established
       Vagrant instance <ansible-centos-70> created.
       Finished creating <ansible-centos-70> (0m3.74s).
-----> Converging <ansible-centos-70>...
       Preparing files for transfer
       Preparing playbook
[...]
      PLAY [tomcat-servers] ********************************************************* 
       
       GATHERING FACTS *************************************************************** 
       <172.28.128.7> ESTABLISH CONNECTION FOR USER: vagrant
       <172.28.128.7> REMOTE_MODULE setup
       <172.28.128.7> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o Port=22 -o IdentityFile="/home/vagrant/.ssh/tomcat_private_key.pem" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=10 172.28.128.7 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1455066081.77-139647883634890 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1455066081.77-139647883634890 && echo $HOME/.ansible/tmp/ansible-tmp-1455066081.77-139647883634890'
       EXEC previous known host file not found for 172.28.128.7
       fatal: [172.28.128.7] => SSH Error: ssh: connect to host 172.28.128.7 port 22: Connection timed out
           while connecting to 172.28.128.7:22
       It is sometimes useful to re-run the command using -vvvv, which prints SSH debug output to help diagnose the issue.
       
       TASK: [selinux | Download EPEL Repo - Centos/RHEL 6] ************************** 
       FATAL: no hosts matched or all hosts have already failed -- aborting
       
       
       PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/root/default.retry
       
       172.28.128.7               : ok=0    changed=0    unreachable=1    failed=0
       
>>>>>> Converge failed on instance <ansible-centos-70>.

```
it seems to launch ansible on corresponding server before initialization of server

There is a test-kitchen bug relative to multi-instance
https://github.com/test-kitchen/test-kitchen/issues/184
not sure if it is the most relevant. would it be appropriate to up it?

If I do one by one, it's working ... but not fully automated test now
```
$ kitchen create ansible-centos-70
$ kitchen create tomcat-centos-70
$ cp .kitchen/kitchen-vagrant/kitchen-ansible_repo-tomcat-centos-70/.vagrant/machines/default/virtualbox/private_key spec/tomcat_private_key.pem
$ kitchen converge ansible-centos-70
$ kitchen verify ansible-centos-70
```

If do my targeted role with windows in manual mode
```
$ kitchen create ansible-centos-70
$ kitchen create windows-windows-2012r2
$ kitchen converge ansible-centos-70
[...]
       Thank you for installing Chef!
       Transferring files to <ansible-centos-70>
       Going to invoke ansible-playbook with: ANSIBLE_ROLES_PATH=/tmp/kitchen/roles sudo -Es  ansible-playbook --inventory-file=/tmp/kitchen/hosts -c winrm -M /tmp/kitchen/modules         /tmp/kitchen/config.yml
       ERROR: pattern is not a legal parameter of an Ansible Play
>>>>>> Converge failed on instance <ansible-centos-70>.
$ egrep -v '(^#|^$)' .kitchen.yml 
---
driver:
  name: vagrant
provisioner:
  name: ansible_playbook
  roles_path: ../../roles
  hosts: test-kitchen
  ansible_verbose: false
  ansible_verbosity: 3
  ansible_extra_flags: <%= ENV['ANSIBLE_EXTRA_FLAGS'] %>
  ansible_yum_repo: http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
  enable_yum_epel: true
  ansible_connection:  winrm
  require_windows_support: true  
platforms:
  - name: centos-7.0
    driver_plugin: vagrant
    driver_config:
      box: centos-7.0
  - name: windows-2012r2
    transport:
      name: winrm
    driver:
      gui: false
suites:
  - name: ansible
    run_list:
    attributes:
    driver_config:
      network:
      - [ 'private_network', { ip: '172.28.128.6' } ]
      - [ 'public_network', { bridge: 'eth0' }]
    includes:
      - centos-7.0
    provisioner:
      name: ansible_playbook
      playbook: test/integration/ansible/ansiblespec/config.yml
  - name: windows
    driver_config:
      network:
      - [ 'private_network', { ip: '172.28.128.7' } ]
    includes:
      - windows-2012r2
    provisioner:
      name: ansible_playbook
      playbook: test/integration/default/default.yml
```

after some changes
```
$ kitchen create ansible-centos-70
$ kitchen create windows-windows-2012r2
$ kitchen converge ansible-centos-70
[...]
       Transferring files to <ansible-centos-70>
       Going to invoke ansible-playbook with: ANSIBLE_HOST_KEY_CHECKING=false ANSIBLE_ROLES_PATH=/tmp/kitchen/roles sudo -Es  ansible-playbook --inventory-file=/tmp/kitchen/hosts -c winrm -M /tmp/kitchen/modules         /tmp/kitchen/default.yml
       
       PLAY [all] ******************************************************************** 
       
       GATHERING FACTS *************************************************************** 
       fatal: [172.28.128.7] => 500 WinRMTransport. [Errno 113] No route to host

$ kitchen list
Instance                Driver   Provisioner      Verifier  Transport  Last Action
ansible-centos-70       Vagrant  AnsiblePlaybook  Busser    Ssh        Converged
windows-windows-2012r2  Vagrant  AnsiblePlaybook  Busser    Winrm      Created
$ kitchen login ansible-centos-70
Last login: Wed Feb 10 02:39:03 2016 from 10.0.2.2
[vagrant@ansible-centos-70 ~]$ ifconfig 
enp0s3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.2.15  netmask 255.255.255.0  broadcast 10.0.2.255
        inet6 fe80::a00:27ff:fec3:36f0  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:c3:36:f0  txqueuelen 1000  (Ethernet)
        RX packets 880  bytes 99753 (97.4 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 598  bytes 87671 (85.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

enp0s8: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.28.128.6  netmask 255.255.255.0  broadcast 172.28.128.255
        inet6 fe80::a00:27ff:fe7c:fab3  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:7c:fa:b3  txqueuelen 1000  (Ethernet)
        RX packets 33  bytes 4657 (4.5 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 35  bytes 5308 (5.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

enp0s9: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.2.123  netmask 255.255.255.0  broadcast 192.168.2.255
        inet6 fe80::a00:27ff:fe4c:7632  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:4c:76:32  txqueuelen 1000  (Ethernet)
        RX packets 163  bytes 19153 (18.7 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 66  bytes 10099 (9.8 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```
IP on enp0s3 is found identical on ansible and windows box
enp0s8(.6) is set correctly and windows side (.7) too but they can't ping each other and don't see anything with tcpdump
=> vagrant network problem or kitchen? or just because centos is epel and ansible1.9?

If I execute my role from host to another win VM, it's fine.

I switched kitchen ansible controller to ubuntu with ansible 2.0. For windows, that's changing a lot. But had to manually install python-pip, 'pip install xmltodict', 'pip install https://github.com/diyan/pywinrm/archive/master.zip#egg=pywinrm'.
I'm a bit surprised that this was not an issue on centos, especially pywinrm...
but get a different error
```
$ kitchen verify
-----> Starting Kitchen (v1.4.2)
-----> Converging <ansible-ubuntu-1404>...
       Preparing files for transfer
[...]
       Installing ansible, will try to determine platform os
       Transferring files to <ansible-ubuntu-1404>
       Going to invoke ansible-playbook with: ANSIBLE_HOST_KEY_CHECKING=false ANSIBLE_ROLES_PATH=/tmp/kitchen/roles sudo -Es  ansible-playbook --inventory-file=/tmp/kitchen/hosts -c winrm -M /tmp/kitchen/modules         /tmp/kitchen/default.yml
       
       PLAY ***************************************************************************
       
       TASK [setup] *******************************************************************
       An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: argument of type 'NoneType' is not iterable
       fatal: [172.28.128.7]: FAILED! => {"failed": true, "stdout": ""}
       
       NO MORE HOSTS LEFT *************************************************************

```

:(

To be continued

**Comment by neillturner** [comment]
I will test myself when i get time. You probably need to go onto the ansible and write a little python script to winrm to the windows server.

**Comment by juju4** [comment]
Ok got it. Was believing packer had include ansible support but not (after all page, is chef-focused).
Removed windows firewall and ping works
Execute ConfigureRemotingForAnsible.ps1 and win_ping and the rest work

added a note on https://github.com/boxcutter/windows/issues/20

more manual stuff. will review that.

**Comment by juju4** [comment]
Didn't manage to get it work for now. back at network issue.
I uploaded my test repository if you want to check my config. README.md has my current kitchen test run.
https://github.com/juju4/ansible-loki

Normal execution as a roles is working both on linux and windows.

**Comment by neillturner** [comment]
windows support in now available in version 0.40.0
also dynamic inventories support was added in vesion 0.39.0
closing this issue. If you have issues running the ansible jinj2 templates then create a separate issue.
